src/Util/CloneTrait.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Util;
+
+/**
+ * Clone given data if cloneable.
+ *
+ * @internal
+ *
+ * @author Quentin Barloy <quentin.barloy@gmail.com>
+ */
+trait CloneTrait
+{
+    public function clone($data): bool
+    {
+        try {
+            return \is_object($data) && (new \ReflectionClass(\get_class($data)))->isCloneable() ? clone $data : $data;
```

Comment:
We should return `null` instead of the updated data if not cloneable (same in the exception).